src/background/Wallet/Wallet.ts

```diff
@@ -799,6 +805,29 @@ export class Wallet {
     chainConfigStore.setDefiSdkClient(client);
   }
 
+  private updateLastAddress() {
+    const currentAddress = this.readCurrentAddress();
+    if (this.id && currentAddress) {
+      lastUsedAddress.setState({
+        address: currentAddress,
+        walletModelId: this.id,
+      });
+    }
+  }
+
+  async getLastUsedAddress({
+    context,
+    params: { userId },
+  }: WalletMethodParams<{ userId: string }>) {
+    this.verifyInternalOrigin(context);
+    const state = lastUsedAddress.getState();
+    if (state && state?.walletModelId === userId) {
```

Comment:
Maybe I don't get the idea but from the first sight it is not clear why are we comparing `walletModelId` to `userId`